catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/formulaeditor/FormulaEditorDiscardChangesTest.java

```diff
@@ -0,0 +1,161 @@
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.uiespresso.formulaeditor;
+
+import android.support.test.runner.AndroidJUnit4;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.content.Script;
+import org.catrobat.catroid.content.bricks.PlaceAtBrick;
+import org.catrobat.catroid.ui.SpriteActivity;
+import org.catrobat.catroid.uiespresso.content.brick.utils.BrickTestUtils;
+import org.catrobat.catroid.uiespresso.testsuites.Cat;
+import org.catrobat.catroid.uiespresso.testsuites.Level;
+import org.catrobat.catroid.uiespresso.util.rules.BaseActivityInstrumentationRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.Espresso.pressBack;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickDataInteractionWrapper.onBrickAtPosition;
+import static org.catrobat.catroid.uiespresso.formulaeditor.utils.FormulaEditorWrapper.onFormulaEditor;
+import static org.catrobat.catroid.uiespresso.util.UiTestUtils.getResourcesString;
+import static org.catrobat.catroid.uiespresso.util.UiTestUtils.onToast;
+
+@RunWith(AndroidJUnit4.class)
+public class FormulaEditorDiscardChangesTest {
+
+	private int brickPosition;
+	private String no = getResourcesString(R.string.no);
+	private String yes = getResourcesString(R.string.yes);
+
+	@Rule
+	public BaseActivityInstrumentationRule<SpriteActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(SpriteActivity.class, SpriteActivity.EXTRA_FRAGMENT_POSITION, SpriteActivity.FRAGMENT_SCRIPTS);
+
+	@Before
+	public void setUp() throws Exception {
+		brickPosition = 1;
+		Script script = BrickTestUtils.createProjectAndGetStartScript("FormulaEditorDiscardChangesTest");
+		script.addBrick(new PlaceAtBrick());
+		baseActivityTestRule.launchActivity();
+	}
+
+	@Category({Cat.AppUi.class, Level.Smoke.class})
+	@Test
+	public void testGoBackToDiscardXChanges() {
+		onBrickAtPosition(brickPosition).checkShowsText(R.string.brick_place_at);
+		onBrickAtPosition(brickPosition).onFormulaTextField(R.id.brick_place_at_edit_text_x)
+				.perform(click());
+		onFormulaEditor()
+				.performEnterFormula("1234");
+		pressBack();
+		onView(withText(R.string.formula_editor_discard_changes_dialog_title))
+				.check(matches(isDisplayed()));
+		onView(withText(no))
```

Comment:
it irritates me, that the tests clicks 'no' = 'do not discard' and the positions are nevertheless set back to default zero. Or do I get this wrong?

Reply:
"no" means do not save changes.
thats ok, the ticket is about bugs when theres formulas not correctly set back to initial state, etc...

Reply:
https://github.com/Catrobat/Catroid/blob/b7cdf2126c624e71b8aa543288775cc9b0b24358/catroid/src/main/res/values/strings.xml#L1885
See string text/value (vs. name).

Reply:
Ömer and me agreed on renaming the variable 'R.string....discard...' in 'R.string....save...' because this is also the title of the dialog.